src/engine/renderer.js

```diff
@@ -260,11 +315,11 @@ export const render = (sceneA, sceneB, mixRatio, viewerMode, deltaIn) => {
       outputCanvas.style.display = 'block'
 
       // Render for output
-      renderLogic(sceneA, sceneB, mixState, mixRatio)
+      renderLogic(mixState, mixRatio)
       // Copy pixels to output canvas
       copyPixels(outputContext)
       // Render for preview
-      renderLogic(sceneA, sceneB, viewerMode, mixRatio)
+      renderLogic(viewerMode, mixRatio)
```

Comment:
I wouldn't let this block this branch, as it's the same amount of computation as before. But I'm wondering your thoughts on scaling down the output texture and rendering that to the preview @funwithtriangles ? I feel that would save processing when you have both the output and preview running during a show, preview might look a little chunky but I don't think that's really a big deal

Reply:
I think copying the pixels is actually faster, no?

Reply:
yeah, but renderLogic gets called twice in this branch of the if statement, once for the preview and once for the output

Reply:
Heh, this is actually a bit confusing. This block is for when you have the viewer set to something different than the output. For example, you may be previewing channelA but the audience is seeing the mix, which is set to channelB, therefore two renders! Not actually a feature I ever use but I think it's a nice thing!

Reply:
The nice thing is, that if you have viewer mode set to 'A' and the mix is all the way over to 'A', it will just render once.

Reply:
However I guess in this instance it would make sense to render the preview at a much smaller resolution.

Reply:
I've made an issue for it https://github.com/nudibranchrecords/hedron/issues/352

Reply:
Oh, I thought the two calls were rendering the same scene, looks like you gathered a different improvement though, haha.

I didn't know you could set the viewer to something else, I'll have to play around with that!